### PR TITLE
Lint no bind

### PR DESCRIPTION
**Pull request by atravkovs**
Refactored code to match `react/jsx-no-bind`. Intentionally skipped `MyAccountDetails.component.js` as this functionality is going to be changed in the nearest future anyway (as per Roadmap).

Fixes: https://github.com/scandipwa/base-theme/issues/240

### CODE REVIEW

File: src/app/component/ReviewStar/ReviewStar.style.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+.ReviewStar {
```
[the remark this answers is not in the document]
Comment (xapiksw) [reply src/app/component/ReviewStar/ReviewStar.style.scss]: Have left to ease extensibility, as adding special styles were preserved in previous version. If not the case can be safely removed.

**Comment by alfredsgenkins** [comment]
@xapiksw Why is some `CategoryPaginationLink` things here? Intentional?

**Comment by atravkovs** [comment]
> @xapiksw Why is some `CategoryPaginationLink` things here? Intentional?

Yes, they are intentional, to get rid of function recreation inside render function. More info about approach can be found here: [`rules/jsx-no-bind` List of items tip](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#lists-of-items)

**Comment by alfredsgenkins** [comment]
We must use either `onClick: () => {}` either `.bind()` in constructor. We now have a mix of them. So either bind, either arrow functions.

**Pull request merged**